smoke-tests/spec/external_dns_spec.rb

```diff
@@ -1,39 +1,44 @@
 require "spec_helper"
 
 describe "external DNS" do
-  # let!(zone) { nil }
+  namespace = "child-#{readable_timestamp}"
+  let(:domain) { "child.parent.service.justice.gov.uk" }
+  let(:ingress_domain) { domain }
+  let(:ingress_name) { domain }
+  let(:parent_domain) { "parent.service.justice.gov.uk" }
+  let(:fixture_name) { "spec/fixtures/external-dns-ingress.yaml.erb" }
+  zone = nil
+  parent_zone = nil
 
   context "when zone matches ingress domain" do
-    let(:domain) { "child.parent.service.justice.gov.uk" }
-    let(:namespace) { "child-parent" }
-    let(:ingress_domain) { domain }
-    let(:parent_zone_id) { "ZQVC43X15AWL9" }
-    zone = nil
-
     # Create a new zone per test
     before do
+      parent_zone = create_zone(parent_domain)
       zone = create_zone(domain)
-      create_delegation_set(zone, parent_zone_id)
+      create_delegation_set(zone, parent_zone.hosted_zone.id)
     end
 
     after do
-      cleanup_zone(zone, domain)
-      delete_delegation_set(zone, parent_zone_id)
+      cleanup_zone(zone, domain, namespace, ingress_name)
+      delete_delegation_set(zone, parent_zone.hosted_zone.id)
+      delete_zone(parent_zone.hosted_zone.id)
     end
 
     # When I create an ingress
     context "when an ingress is created" do
-      before do
+      before(:all) do
+        sleep 1
         create_namespace(namespace)
       end
 
-      after do
+      after(:all) do
         delete_namespace(namespace)
       end
 
       # an A record should be created
       it "creates an A record" do
-        create_ingress
+        create_ingress(namespace, ingress_name, fixture_name)
```

Comment:
Given that the context name mentions that an ingress is created, I would move the `create_ingress` call, and the associated sleep, into the `before(:all)` block